wsl-pro-service/internal/wslinstanceservice/wslinstanceservice.go

```diff
@@ -88,19 +89,20 @@ func (s *Service) sendInfo(ctx context.Context) error {
 	}
 
 	if err := s.ctrlStream.Send(sysinfo); err != nil {
-		return fmt.Errorf("could not send back system info: %v", err)
+		return fmt.Errorf("could not send system info: %v", err)
 	}
 
 	return nil
 }
 
 // ApplyLandscapeConfig serves LandscapeConfig messages sent by the agent.
 func (s *Service) ApplyLandscapeConfig(ctx context.Context, msg *wslserviceapi.LandscapeConfig) (empty *wslserviceapi.Empty, err error) {
-	defer decorate.OnError(&err, "ApplyLandscapeConfig error")
+	defer decorate.LogOnError(err)
+	defer decorate.OnError(&err, "ApplyLandscapeConfig")
```

Comment:
I find this chaining quite weird (return an error + logging is generally not a good pattern resulting in stuttering).

Also, on the error itself, you are prefixing here with the funciton name, that you are removing on other error log. I think we should be consistent

Reply:
I only Log+return in one case: gRPC methods.

This is because this error is never printed, just sent back via gRPC (I think, not sure where it goes). The proper way of doing this would be with an interceptor, but his seemed like a quick-and-easy way to at least get the logs.

The inconsistency was accidental.

Reply:
Ok, I think then we should log in a middleware when we return an error from the grpc server, wdyt?

Reply:
I agree, you caught me trying go get away with the lazy solution 😅. I'll try setting this up. Won't be too different from adsys I figure.